tests/e2e/tests/e2e_happy_path/HappyPath.spec.ts

```diff
@@ -176,20 +176,18 @@ suite('Validation of workspace build and run', async () => {
 
     test('Run application', async () => {
         await topMenu.runTask('run');
-        await ide.waitNotificationAndConfirm('A new process is now listening on port 8080', 120_000);
-        applicationUrl = await ide.getApplicationUrlFromNotification('Redirect is now enabled on port 8080', 120_000);
-        await ide.waitNotificationAndOpenLink('Redirect is now enabled on port 8080', 120_000);
+        await ide.waitNotification('Process 8080-tcp is now listening on port 8080. Open it ?', 120_000);
+        await driverHelper.wait(TimeoutConstants.TS_SELENIUM_PREVIEW_WIDGET_DEFAULT_TIMEOUT);
```

Comment:
User the simple `wait` is not a best practice for tests, ti should always use the flow `wait for something`. Can you explain why is this (and two more waits later on lines 22 and 247) added to the code?

Reply:
Yes , it's a not the best practise.. but we want reliable E2E test.
When I was checkeing the changes launching the `happy-path` , I got expected view of demo application not always on this step, it needed some timeout after appereance the notification and before click on the button 'Open In Preview'.. and nothing to wait - the notification is present, the button is available.. 'selenium driver' works too fast and spoils our life sometimes :)

Reply:
Ok, thanks for the answer. It would be great to avoid this approach if possible but I understand your case here. Thanks for checking it again in future :)